### ISSUE TITLE
Warning about `ref` when using another HOC as a custom input

### ISSUE THREAD

**Issue by pandaiolo** [s-yadav/react-number-format#136]
Related to #125 but not the same issue. #125 is about getting the inputRef from the component. This issue is about a warning that is shown when using an HOC as material-ui `TextField` as a custom input of `react-number-format`.

Received warning:
```
Warning: Stateless function components cannot be given refs. Attempts to 
access this ref will fail.

      Check the render method of `NumberFormat`.
          in TextField (created by NumberFormat)
          in NumberFormat (created by MyApp)
          in MyApp
          in WrapperComponent
```

I think the problem is the following:
- `<NumberFormat />` use `ref` pseudo prop on the custom input to get a reference to it
- Instead, `<TextField/>` asks to use` inputRef` to get underlying input element
- Result: `<NumberFormat />` does not get `<TextField/>` `<input/>` ref

Workaround I used is to use a HOC that forwards input`ref` to `<TextField/>` ` inputRef` as pictured below:

```js
import React, { createElement } from 'react'
import TextField from 'material-ui/TextField'
import NumberFormat from 'react-number-format'
import createWithRef from 'react-hoc-with-ref'

class CustomInput extends createWithRef(TextField) {
  render = () =>
    createElement(TextField, {
      ...this.props,
      inputRef: this.getRefProps().ref
    })
}

const MyApp = () => 
    <NumberFormat customInput={CustomInput} /* ... other props */ />
```

What do you think of this workaround? 

Anybody having ref issue and trying this, let me know your experience!

Edit: I think it would work with #131 as well.

**Comment by s-yadav** [comment]
Doesn't this work. https://github.com/s-yadav/react-number-format#getting-reference

As NumberFormat forwards all the extra props to the CustomInput you can directly pass inputRef prop
```jsx
<NumberFormat inputRef={(elem) => this.elem = elem} customInput={TextField} /* ... other props */ />
```

**Comment by pandaiolo** [comment]
Hum yes, If I myself want a ref I can use that. My issue was more about the internals of `<NumberFormat>`. Although, now that I read the code more thoroughly, I see that the `ref` prop passed to the input is just the `getInputRef` from its own props. `NumberFormat` itself is not using the input's ref... 

Sooooo, I should not need that HOC in the end? But why am I getting the above warning? If I don't set `getInputProps`, underlying `ref` should also be undefined...

**Comment by s-yadav** [comment]
Ohh got it,
https://github.com/s-yadav/react-number-format/blob/master/src/number_format.js#L768 .
ref should not be added there, As that is passed to both customInput or normal input. 
I will push the fix.

**Comment by s-yadav** [comment]
Fixed this on v3.1.1 
Let me know if this is still coming

**Comment by pandaiolo** [comment]
If anybody use a custom input that uses `ref`, they will not be able to reference this custom input through `getInputRef`.

Well, it is a custom input, if they need a ref they can wrap it in a HOC with custom ref prop before giving it to NumberFormat :)

Thanks for the fix, I confirm it works!
